docs/design/windows-design.md

```diff
@@ -108,94 +108,74 @@ The packets that are sent to/from the Windows host should be forwarded on this i
-addresses. It is implemented using OpenFlow.
-
-To support this feature, two additional marks are introduced:
-
-* The 17th bit of NXM Register0 is set for Pod-to-external traffic. This bit is set in the `L3Forwarding` table,
- and is consumed in the `ConntrackCommit` table.
-* The SNATed traffic is marked with **0x40** in the ct context. This mark is set in the `ConntrackCommit`
- table when committing the new connection into the ct zone, and is consumed in the `ConntrackState` table
- to identify SNAT relevant packets.
-
-The following changes in the OVS pipeline are needed:
-
-* Identify the packet that is sent from local Pod-to-external address and set the SNAT bit in the register
- in the `L3Forwarding` table. The packet should have these characteristics:
-  1) it enters the OVS pipeline from a local Pod;
-  2) the destination IP address is not in the cluster Pod CIDRs (Pod subnets of the local or any remote
-  Node), and is not the local Node's management IP;
-  3) the traffic is not for Services;
-  4) it is not the return traffic for a connection from an external IP to a local Pod.
-* Implement SNAT rules and commit the new connection to the ct context in the `ConntrackCommit` table. The SNAT
- rule should change the source address to the Node's internal IP.
-* Forward the IP packet to the `Conntrack` table if it is from the uplink interface in the `Uplink` table.
- This is to ensure the reply packets from the external address to the Pod will be "unSNAT'd".
-* Ensure that the packet is translated based on the NAT information provided when committing the connection by
- using the `nat` argument with the `ct` action.
-* Set the macRewrite bit in the register (reg0[19]) in the `ConntrackState` table if the packet is from the uplink
- interface and has the SNAT ct_mark.
-
-Following is an example for SNAT relevant OpenFlow entries.
+Following are the OpenFlow entries for uplink interface in encap mode.
 
 ```text
 Classifier Table: 0
-table=0, priority=200, in_port=$uplink,ip actions=load:0x4->NXM_NX_REG0[0..15],goto_table:5
-
-Uplink Table: 5
-table=5, priority=200, ip,reg0=0x4/0xffff actions=goto_table:30
+table=0, priority=200, in_port=$uplink actions=LOCAL
+table=0, priority=200, in_port=LOCAL actions=output:$uplink
+```
 
-Conntrack Table: 30
-table=30, priority=200, ip actions=ct(table=31,zone=65520,nat)
+Following is an example for the OpenFlow entries related with uplink interface in noEncap mode.
 
-ConntrackState Table: 31
-table=31, priority=210, ct_state=-new+trk,ct_mark=0x40,ip,reg0=0x4/0xffff actions=load:0x1->NXM_NX_REG0[19],goto_table:40
-table=31, priority=200, ip,reg0=0x4/0xffff actions=output:br-int
+```text
+Classifier Table: 0
+table=0, priority=200, ip, in_port=$uplink, nw_dst=$localPodSubnet, actions=load:0x4->NXM_NX_REG0[0..15],load:0x
+1->NXM_NX_REG0[19],resubmit(,29)
+table=0, priority=200, in_port=$uplink actions=LOCAL
```

Comment:
I think the above two flows shouldn't have same priority? otherwise it's not predictable for the packet matching both.